front/src/components/App/MaintenanceArea/Utils.tsx

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import {AuctionSite} from "./AuctionSitesManagement";
+
+const existsThisName = (auctionSites: AuctionSite[], item: AuctionSite) => {
+  if (!auctionSites || item.name === "")
+    return true
+  for (let i = 0; i < auctionSites.length; i++) {
+    if (auctionSites[i].name === item.name) {
+      return true
+    }
+  }
+  return false
+}
+
+export function addAuctionSite(auctionSites: AuctionSite[], item: AuctionSite) {
+  if (!existsThisName(auctionSites, item)) {
+    if (auctionSites) {
+      auctionSites = [...auctionSites, item]
```

Comment:
BARDZO zła praktyka mutowania propsów. Wszystko przez to że te funkcje zostały przeniesione do innego pliku. Zamiast tego powinny być umieszczone tam gdzie są wykonywane. Komponent dodający do listy powinien wiedzieć jak do tej listy dodawać